### DISCUSSION TITLE
All extension settings dialogs crash after a few seconds (seems to happen on Arch after Gnome 45.3)

### DISCUSSION THREAD

**Discussion by jtaala** [paperwm/PaperWM#749] [Q&A] [answered]
Hey @Lythenas, I know you're also on Arch.  Are you seeing this?

All extension settings dialogs crash for me after some seconds (3-8 seconds type thing).  At first I thought it was paperwm dialog but happens with other dialogs as well (also happens with no extensions enabled).

Found another Arch/nvidia user that was seeing this too:
https://bbs.archlinux.org/viewtopic.php?pid=2142864#p2142864

Let me know if you're seeing this.

Cheers!

**Comment by Lythenas** [comment]
Just updated my system (gnome and mutter 45.2 -> 45.3 and a bunch of other probably unrelated stuff) and it also crashes now. I also can't open the settings again once they crashed once.

When I try to open the settings again (though the topbar) I get this in the logs:

```
gnome-shell[1638]: Unhandled promise rejection. To suppress this warning, add an error handler to your promise chain with .catch() or a try-catch block around your await expression. Stack trace of the failed promise:
		_promisify/proto[asyncFunc]@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:448:16
		openExtensionPrefs@resource:///org/gnome/shell/ui/extensionSystem.js:308:26
		openPreferences@resource:///org/gnome/shell/extensions/extension.js:27:26
		openPrefs@file:///home/ms/.local/share/gnome-shell/extensions/paperwm@paperwm.github.com/topbar.js:40:33
		_init/<@file:///home/ms/.local/share/gnome-shell/extensions/paperwm@paperwm.github.com/topbar.js:470:17
		activate@resource:///org/gnome/shell/ui/popupMenu.js:194:14
		_init/<@resource:///org/gnome/shell/ui/popupMenu.js:111:24
		@resource:///org/gnome/shell/ui/init.js:21:20
```

When I try to open it through the gnome extensions app I get this:

```
gjs[7182]: JS ERROR: Gio.IOErrorEnum: GDBus.Error:org.gnome.gjs.JSError.Error: Already showing a prefs dialog
                                   asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:114:23
                                   _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
```

but there is definitively no dialog showing.

Also dialogs for all other extensions are the same.

**Comment by jtaala** [comment]
Yep, can see issue here.  Looks like an Arch only issue atm (or at least haven't seen posts from other distro users seeing this):

https://bbs.archlinux.org/viewtopic.php?pid=2142972#p2142972

**Comment by jtaala** [comment]
Others have reported that opening through `extension manager` or the `extension list` extension results in no crash.  However, it's crashing for me no matter what method I use to open settings in PaperWM.

You can kill gjs extension process when to open the extension settings again (without an logout):

```
killall gjs
```

**Comment by jtaala** [comment, marked as answer]
It's been reported upstream so hopefully a resolution will be coming.

https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7325

**Reply by Thesola10** [reply]
Fix has been accepted and landed on Arch repos as `1.78.3`, affected users should upgrade their system.

**Comment by jtaala** [comment]
Hey @Lythenas,

See the latest comment(s) on https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7325.  This issue is caused by `gjs` version 1.78.2.

I've just downgraded my `gjs` package back to `1.78.1` and this issue no longer occurs (and I can again change my prefs settings).  You can install the [downgrade](https://aur.archlinux.org/packages/downgrade) AUR package and then do a:

```
sudo downgrade gjs
```
and select version `1.78.1`.  I'll stay on `1.78.1` until they release a fix for it.

**Reply by Lythenas** [reply]
Can confirm this works.

**Reply by jtaala** [reply]
Note too @Lythenas, if you used `downgrade` you'd want to remove any `ignorePkg` entry for `gjs` in `/etc/pacman.conf`  (downgrade usually adds this for you so your downgraded package isn't automatically updated again).
